app/js/sumo_db.js

```diff
@@ -31,6 +19,8 @@
       req.onload = function() {
         if (req.status >= 200 && req.status < 300) {
           resolve(req.response);
+        } else if (req.status == 400) {
+          reject(req.response);
```

Comment:
We're not catching any failed promise at the moment ([but we will](https://bugzilla.mozilla.org/show_bug.cgi?id=1128547)) so I think it's good to just do this all the time:
```js
} else {
  reject(req.response);
}
```